proof/aggregate_in_g1.saw

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2020 Galois, Inc.
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+*/
+
+// NOTE: we only consider inputs that either have the infinity-bit set or compressed-bit set
+
+// aggregate_in_g1_error_precond describes the circumstances under which aggregate_in_g1 returns an error (assuming either the compressed bit is set of the infinity bit is set)
+let {{
+  aggregate_in_g1_error_precond : [48][8] -> Bool
+  aggregate_in_g1_error_precond zwire =
+      ~(zwire@0)@1 // not infinity
+    /\ (   uncompress_E1_x zwire >= `p
+        \/ ~ is_square_fp (uncompress_E1_y2 zwire) // not on curve
+        \/ uncompress_E1_x_fp zwire == Fp.field_zero // uncompresses to zero
+        \/ ~ is_in_g1_impl (uncompress_E1_OK zwire) ) // not in group g1
+}};
+
+// case in which the input is not null and we return BLST_SUCCESS:
+let blst_aggregate_in_g1_OK_nonnull_spec = do {
+  //
+  (out, out_ptr) <- ptr_to_fresh "blst_aggregate_in_g1_out" (llvm_struct "struct.POINTonE1");
+  (inp, inp_ptr) <- ptr_to_fresh_readonly "blst_aggregate_in_g1_in" (llvm_struct "struct.POINTonE1");
+  (zwire, zwire_ptr) <- ptr_to_fresh_readonly "blst_aggregate_in_g1_zwire" (llvm_array 48 (llvm_int 8));
+  //
+  llvm_precond {{ (zwire@0)@0 }}; // compressed
+  llvm_precond {{ POINTonE1_invariant inp }}; // inp is a valid point (precondition of add)
+  llvm_precond {{ ~aggregate_in_g1_error_precond zwire }}; // no error condition
+  //
+  llvm_execute_func [out_ptr, inp_ptr, zwire_ptr];
+  //
+  new_out <- llvm_fresh_var "new_out" (llvm_struct "struct.POINTonE1");
+  llvm_points_to out_ptr (llvm_term new_out);
+  llvm_postcond {{ affinify E (POINTonE1_abs new_out) == if (zwire@0)@1 then affinify E (POINTonE1_abs out) else add E (affinify E (POINTonE1_abs inp)) (uncompress_E1_OK zwire) }};
+  //
+  llvm_return (llvm_term {{ (`BLST_SUCCESS):[32] }});
+};
+
+// Special case: the input is null:
+
+let blst_aggregate_in_g1_null_spec = do {
+  //
+  (out, out_ptr) <- ptr_to_fresh "blst_aggregate_in_g1_out" (llvm_struct "struct.POINTonE1");
```

Comment:
Here it definitely seems unnecessary to point to initialized memory; out_ptr is always written to.

Reply:
Fixed but moved to new pull request